### PR TITLE
logging support: optimize escaping: no regex

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by protocol7** [comment]
Looks like ident escaping is missing tests, perhaps worth adding as part of the commit. Other than that, +1

**Comment by danielnorberg** [comment]
Good catch. Added a test in a commit before the optimization commit to verify that the ident escaping behavior does not change.

File: src/main/java/com/spotify/logging/LoggingSupport.java [hunk 1]
```diff
@@ -149,16 +149,25 @@ protected static String buildLogLine(final String type, final int version, final
...
-  protected static String escape(final Object o) {
+  protected static void appendEscaped(final Object o, final StringBuilder out) {
     if (o == null) {
-      return "";
+      return;
+    }
+    final String s = o.toString();
+    for (int i = 0; i < s.length(); i++) {
+      final char c = s.charAt(i);
+      if (c == '\t' || c == '\n') {
```
Comment (spkrka) [inline src/main/java/com/spotify/logging/LoggingSupport.java]: could do `c < ' '` i guess - are there any <32 characters worth keeping?

**Comment by spkrka** [comment]
:+1:

File: src/main/java/com/spotify/logging/LoggingSupport.java (hunk 1 above)
Comment (danielnorberg) [inline src/main/java/com/spotify/logging/LoggingSupport.java]: Not sure, didn't want to change behavior.

**Pull request merged**